go/vt/vtgate/vtgateconn/vtgateconn.go

```diff
@@ -18,23 +18,32 @@ package vtgateconn
 
 import (
 	"context"
-	"flag"
 	"fmt"
 	"sync"
 
+	"github.com/spf13/pflag"
+
 	"vitess.io/vitess/go/sqltypes"
 	"vitess.io/vitess/go/vt/log"
+	"vitess.io/vitess/go/vt/servenv"
 
 	binlogdatapb "vitess.io/vitess/go/vt/proto/binlogdata"
 	querypb "vitess.io/vitess/go/vt/proto/query"
 	topodatapb "vitess.io/vitess/go/vt/proto/topodata"
 	vtgatepb "vitess.io/vitess/go/vt/proto/vtgate"
 )
 
-var (
-	// VtgateProtocol defines the RPC implementation used for connecting to vtgate.
-	VtgateProtocol = flag.String("vtgate_protocol", "grpc", "how to talk to vtgate")
-)
+// VtgateProtocol defines the RPC implementation used for connecting to vtgate.
+var VtgateProtocol = "grpc"
```

Comment:
what about unexporting this? (per #10697, "... and add make them package private")

you can see what i did for tabletconn as an example https://github.com/vitessio/vitess/pull/11010

Reply:
https://github.com/vitessio/vitess/pull/11318/commits/7a79b23013c0c75cec03e9ec7712740094d4e466